lib/IRGen/ClassLayout.h

```diff
@@ -143,13 +143,15 @@ class ClassLayout {
              Options.contains(ClassMetadataFlags::ClassHasObjCAncestry));
   }
 
-  bool doesMetadataRequireInitialization() const {
-    return (Options.contains(ClassMetadataFlags::ClassHasResilientAncestry) ||
-            Options.contains(ClassMetadataFlags::ClassHasGenericAncestry) ||
-            Options.contains(ClassMetadataFlags::ClassHasResilientMembers) ||
+  bool doesMetadataRequireUpdate() const {
+    return (Options.contains(ClassMetadataFlags::ClassHasResilientMembers) ||
             Options.contains(ClassMetadataFlags::ClassHasMissingMembers));
   }
 
+  bool doesMetadataRequireInitialization() const {
+    return Options.contains(ClassMetadataFlags::ClassHasGenericAncestry);
```

Comment:
Can you assert `!doesMetadataRequireRelocation()` if you don't want to count those cases?

Reply:
Done, and added comments too!